src/rpg_database/character_sheet/LevelAndXPCrossDependencyCheck.java

```diff
@@ -0,0 +1,98 @@
+package rpg_database.character_sheet;
+
+import java.security.InvalidParameterException;
+import java.util.HashMap;
+
+import rpg_database.character_sheet.exceptions.CrossDependencyException;
+import rpg_database.character_sheet.interfaces.MultipleFieldsGetterSetter;
+
+public class LevelAndXPCrossDependencyCheck implements MultipleFieldsGetterSetter<LevelAndXPCrossDependencyCheck, Integer> {
+
+	private int xp;
+	private int level;
+	private final int maxLevel = 22;
+	private final int maxXp = 70000;
+
+	private final HashMap<Integer, Integer> xpNeedingForLevelMap = fillUpWithLevelXpPairs();
+
+	public LevelAndXPCrossDependencyCheck() {
+		xp = 0;
+		level = 1;
+	}
+
+	public int GetXP() {
+		return xp;
+	}
+
+	@Override
+	public Class<LevelAndXPCrossDependencyCheck> getImplementingClass() {
+		return LevelAndXPCrossDependencyCheck.class;
+	}
+
+	@Override
+	public Class<Integer> getDataTypeClass() {
+		return Integer.class;
+	}
+
+	@Override
+	public void setSelfValueByField(Fields field, Integer value) {

```

Comment:
Ez elég hardcore a szememnek. Nem akarom sokáig húzni, mert ezzel lehet el tudnék egy órát is bíbelődni és vannak fontosabb kérdések. Ha ilyen marad következő körre is akkor megnézem, de megpróbálok élni a feltételezéssel hogy ezt józan önerőből meg fogod változtatni.